cmd/snapctl/main.go

```diff
@@ -57,27 +57,48 @@ func init() {
 	url := flURL.Value
 	ver := flAPIVer.Value
 	secure := false
+	var arg []string
 
 	for idx, a := range os.Args {
 		switch a {
 		case "--url":
-			if err := f1.Parse(os.Args[idx : idx+2]); err == nil {
+			if len(os.Args) >= idx+2 {
+				arg = os.Args[idx : idx+2]
+			} else {
+				break
+			}
+			if err := f1.Parse(arg); err == nil {
```

Comment:
make a func for the similar code so that you won't need to change them all over the place in the future.